probatus/feature_elimination/feature_elimination.py

```diff
@@ -960,23 +1001,28 @@ def _get_feature_shap_values_per_fold(self, X, y, clf, train_index, val_index, s
             (np.array, float, float):
                 Tuple with the results: Shap Values on validation fold, train score, validation score.
         """
-        
+
         # The lightgbm imports are temporarily placed here, until the tests on
         # macOS have been fixed for lightgbm.
-        from lightgbm import early_stopping, LGBMModel
-        
+        from lightgbm import early_stopping, log_evaluation, LGBMModel
+
         X_train, X_val = X.iloc[train_index, :], X.iloc[val_index, :]
         y_train, y_val = y.iloc[train_index], y.iloc[val_index]
 
         fit_params = {
-            'X':X_train,
-            'y':y_train,
-            'eval_set':[(X_val, y_val)],
-            'eval_metric':self.eval_metric
+            'X': X_train,
+            'y': y_train,
+            'eval_set': [(X_val, y_val)],
+            'eval_metric': self.eval_metric
         }
 
         if isinstance(clf, LGBMModel):
-            fit_params['callbacks'] = [early_stopping(self.early_stopping_rounds)]
+            fit_params['callbacks'] = [
+                early_stopping(
+                    self.early_stopping_rounds, first_metric_only=True
+                ),
+                log_evaluation(0),
```

Comment:
How we use verbose parameter now is:
```            verbose (int, optional):
                Controls verbosity of the output:
                - 0 - neither prints nor warnings are shown
                - 1 - 50 - only most important warnings
                - 51 - 100 - shows other warnings and prints
                - above 100 - presents all prints and all warnings (including SHAP warnings).
                - 
  ```

I guess if `self.verbose<50` or `self.verbose<100`  then we should set it to 0, otherwise we can allow for the prints. it might get ugly with logs for this feature if there is anything done in parallel.

What do you think?

Reply:
I think you have a good point. If `self.verbose == 100` it should also do something for the `log_evaluation` callback, but the parameter for `log_evaluation` is the period of logging/messages. so `log_evaluation(1)` will print a message for each validation step, which is indeed very ugly for anything parallel. 

I suggest we do something like this:

```
if isinstance(clf, LGBMModel):
    fit_params['callbacks'] = [
        early_stopping(self.early_stopping_rounds, first_metric_only=True)
    ]
if verbose == 100:
    fit_params['callbacks'].append(log_evaluation(1))
else:
    fit_params['callbacks'].append(log_evaluation(0))
```

So if verbosity is at max, we'll give all the messages, but if verbosity is anything lower we won't.